Backpack-SwiftUI/CardList/Classes/BPKCardList.swift

```diff
@@ -0,0 +1,131 @@
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+public struct BPKCardList<Element: Identifiable, Content: View>: View {
+    private let title: String
+    private let description: String?
+    private let layout: BPKCardListLayout
+    private let initiallyShownCards: Int
+    private let elements: [Element]
+    private let cardForElement: (_ element: Element) -> Content
+    @State private var showingAllCards = false
+
+    public init(
+        title: String,
+        description: String?,
+        layout: BPKCardListLayout,
+        initiallyShownCards: Int = 3,
+        elements: [Element],
+        @ViewBuilder cardForElement: @escaping (_: Element) -> Content
+    ) {
+        self.title = title
+        self.description = description
+        self.layout = layout
+        self.initiallyShownCards = initiallyShownCards
+        self.elements = elements
+        self.cardForElement = cardForElement
+    }
+
+    public var body: some View {
+        switch layout {
+        case .rail(let sectionHeaderButton):
+            railLayout(with: sectionHeaderButton)
+        case .stack(let accessory):
+            stackLayout(with: accessory)
+        }
+    }
+
+    private func railLayout(with sectionHeaderButton: BPKCardListLayout.SectionHeaderAction?) -> some View {
+        cardListSkeleton(with: sectionHeaderButton) {
+            ScrollView(.horizontal) {
+                HStack(spacing: .base) {
+                    ForEach(elements) { index in
+                        cardForElement(index)
+                    }
+                }
+                .padding(.horizontal)
+            }
+        }
+    }
+
+    private func stackLayout(with accessory: BPKCardListLayout.Accessory?) -> some View {
+        cardListSkeleton(with: accessory?.sectionHeaderButton) {
+            let cardsShown = (showingAllCards) ? elements.count : min(initiallyShownCards, elements.count)
+            let filteredCards = elements[0..<cardsShown]
+            ForEach(filteredCards) { element in
+                cardForElement(element)
+            }
+                .padding(.horizontal)
+            accessoryView(accessory: accessory)
+                .padding(.horizontal)
+        }
+    }
+
+    private func sectionHeader(with sectionHeaderButton: BPKCardListLayout.SectionHeaderAction?) -> some View {
+        if let sectionHeaderButton {
+            return BPKSectionHeader(title: title, description: description) {
+                BPKButton(
+                    icon: sectionHeaderButton.icon,
+                    accessibilityLabel: sectionHeaderButton.accessibilityLabel,
+                    action: sectionHeaderButton.action)
+                .buttonStyle((sectionHeaderButton.style == .default) ? .primary : .primaryOnDark)
+            }
+            .padding()
+        } else {
+            return BPKSectionHeader(title: title, description: description)
+                .padding()
```

Comment:
These `.padding` probably need to add a BPKSpacing token to make it explicit. The system default is 16, but not linked to our design system